src/interface/window.c

```diff
@@ -1803,7 +1803,7 @@ void tool_cancel()
 		// Reset map selection
 		RCT2_GLOBAL(RCT2_ADDRESS_MAP_SELECTION_FLAGS, uint16) = 0;
 
-		if (gCurrentToolWidget.widget_index != 0xFFFF) {
+		if (gCurrentToolWidget.widget_index != (rct_widgetindex) 0xFFFF) {
```

Comment:
True, though the solution here is valid for both the current signed and the intended unsigned type. I can remove it from this PR, but the Xcode stuff is still useful.